internal/handlers/equipment_status.go

```diff
@@ -2,109 +2,369 @@ package handlers
+const (
+	EQUIPMENT_UNDER_REPAIR_COMMENT_FOR_ORDER = "Equipment under repair"
+)
+
+func SetEquipmentStatusHandler(logger *zap.Logger, api *operations.BeAPI) {
+	equipmentStatusRepo := repositories.NewEquipmentStatusRepository()
+	orderStatusRepo := repositories.NewOrderStatusRepository()
+
+	statusHandler := NewEquipmentStatus(logger)
+
+	api.EquipmentStatusUpdateEquipmentStatusOnUnavailableHandler = statusHandler.PutEquipmentStatusInRepairFunc(equipmentStatusRepo, orderStatusRepo)
+	api.EquipmentStatusUpdateEquipmentStatusOnAvailableHandler = statusHandler.DeleteEquipmentStatusRemoveFromRepairFunc(equipmentStatusRepo, orderStatusRepo)
+	api.EquipmentStatusCheckEquipmentStatusHandler = statusHandler.GetEquipmentStatusCheckDatesFunc(equipmentStatusRepo)
+	api.EquipmentStatusUpdateRepairedEquipmentStatusDatesHandler = statusHandler.PatchEquipmentStatusEditDatesFunc(equipmentStatusRepo)
 }
 
-type EquipmentStatusName struct {
+type EquipmentStatus struct {
 	logger *zap.Logger
 }
 
-func NewEquipmentStatusName(logger *zap.Logger) *EquipmentStatusName {
-	return &EquipmentStatusName{
+func NewEquipmentStatus(logger *zap.Logger) *EquipmentStatus {
+	return &EquipmentStatus{
 		logger: logger,
 	}
 }
 
-func (c EquipmentStatusName) PostEquipmentStatusNameFunc(repository domain.EquipmentStatusNameRepository) eqStatusName.PostEquipmentStatusNameHandlerFunc {
-	return func(s eqStatusName.PostEquipmentStatusNameParams, access interface{}) middleware.Responder {
+func (c EquipmentStatus) GetEquipmentStatusCheckDatesFunc(
+	eqStatusRepository domain.EquipmentStatusRepository) eqStatus.CheckEquipmentStatusHandlerFunc {
+	return func(s eqStatus.CheckEquipmentStatusParams, access interface{}) middleware.Responder {
 		ctx := s.HTTPRequest.Context()
-		name := s.Name.Name
-		createdStatus, err := repository.Create(ctx, *name)
+		newStatus := s.Name.StatusName
+
+		ok, err := equipmentStatusAccessRights(access)
+		if err != nil {
+			c.logger.Error("Error while getting authorization", zap.Error(err))
+			return orders.NewAddNewOrderStatusDefault(http.StatusInternalServerError).
+				WithPayload(&models.Error{Data: &models.ErrorData{Message: "Can't get authorization"}})
+		}
+
+		if !ok {
+			c.logger.Error("User have no right to check that equipment status has orders for provided dates",
+				zap.Any("access", access))
+			return orders.NewAddNewOrderStatusDefault(http.StatusForbidden).
+				WithPayload(&models.Error{Data: &models.ErrorData{
+					Message: "You don't have rights to update equipment status"}},
+				)
+		}
+
+		if !newStatusIsUnavailable(*newStatus) {
+			c.logger.Error("Wrong new equipment status, status should be only 'not available'", zap.Any("access", access))
+			return orders.NewAddNewOrderStatusDefault(http.StatusForbidden).
+				WithPayload(&models.Error{Data: &models.ErrorData{Message: "Wrong new equipment status, status should be only 'not available'"}})
+		}
```

Comment:
This is the common part of this method and 2 methods below. Move to function to reduce repetition?